components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -220,7 +220,10 @@ public String buildIDToken(OAuthTokenReqMessageContext request, OAuth2AccessToke
             if (authorizationGrantCacheEntry != null) {
                 nonceValue = authorizationGrantCacheEntry.getNonceValue();
                 acrValue = authorizationGrantCacheEntry.getAcrValue();
-                authTime = authorizationGrantCacheEntry.getAuthTime();
+                if (authorizationGrantCacheEntry.getEssentialClaims() != null
```

Comment:
We need to check whether the essential claim is defined for the ID token here. We don't differentiate claims as 'id token' or 'userinfo' when storing them in the cache.